### PR TITLE
BL-4864 look for version embedded in bloomd

### PR DESCRIPTION
**Pull request by JohnThomson**
Corresponding fix for Bloom must be merged first!

---
This change is [Reviewable](https://reviewable.io/reviews/bloombooks/bloomreader/20)

### CODE REVIEW

File: app/src/main/java/org/sil/bloom/reader/WiFi/NewBookListenerService.java [hunk 1]
```diff
@@ -85,7 +74,7 @@ private void listen(Integer port) throws Exception {
         try {
             JSONObject data = new JSONObject(message);
             String title = data.getString("Title");
-            newBookVersion = data.getString("Version");
+            String newBookVersion = data.getString("Version");
```
Comment (hatton) [inline app/src/main/java/org/sil/bloom/reader/WiFi/NewBookListenerService.java]: This is unusual to upper case json property names. https://stackoverflow.com/questions/5543490/json-naming-convention

Is it too late to fix this?

**Comment by hatton** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[app/src/main/java/org/sil/bloom/reader/IOUtilities.java, line 86 at r1](https://reviewable.io:443/reviews/bloombooks/bloomreader/20#-Ks5Sau8G4uqyZyU9Qub:-Ks5Sau8G4uqyZyU9Quc:b-d39078) ([raw file](https://github.com/bloombooks/bloomreader/blob/8dd13ba0af08d9083d79f83cad760eb71e5e67bb/app/src/main/java/org/sil/bloom/reader/IOUtilities.java#L86)):*
> ```Java
>             byte[] buffer = new byte[(int)entry.getSize()];
>             stream.read(buffer);
>             zip.close();
> ```

Does this close need to happen regardless of whether the entry was found, or some other error sent us to catch()?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomreader/20)*

**Comment by JohnThomson** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[app/src/main/java/org/sil/bloom/reader/IOUtilities.java, line 86 at r1](https://reviewable.io:443/reviews/bloombooks/bloomreader/20#-Ks5Sau8G4uqyZyU9Qub:-Ks9KH5cuROcIaEU_TQa:b-896fix) ([raw file](https://github.com/bloombooks/bloomreader/blob/8dd13ba0af08d9083d79f83cad760eb71e5e67bb/app/src/main/java/org/sil/bloom/reader/IOUtilities.java#L86)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

Does this close need to happen regardless of whether the entry was found, or some other error sent us to catch()?
</blockquote></details>

Done.

---

*[app/src/main/java/org/sil/bloom/reader/WiFi/NewBookListenerService.java, line 77 at r1](https://reviewable.io:443/reviews/bloombooks/bloomreader/20#-KrrUoqrFIGl4cti8YYo-r1-77:-Ks9KndElmgTBG5EEltg:b-896fix) ([raw file](https://github.com/bloombooks/bloomreader/blob/8dd13ba0af08d9083d79f83cad760eb71e5e67bb/app/src/main/java/org/sil/bloom/reader/WiFi/NewBookListenerService.java#L77)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

This is unusual to upper case json property names. https://stackoverflow.com/questions/5543490/json-naming-convention

Is it too late to fix this?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomreader/20)*

**Comment by hatton** [comment]
Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[app/src/main/java/org/sil/bloom/reader/IOUtilities.java, line 84 at r2](https://reviewable.io:443/reviews/bloombooks/bloomreader/20#-Ks9jJF1UtPDYh3_A7fj:-Ks9jJF1UtPDYh3_A7fk:bq2m8m9) ([raw file](https://github.com/bloombooks/bloomreader/blob/d35e79051ad8d19cfd8e8e4b7ae9fd7ca4876ba6/app/src/main/java/org/sil/bloom/reader/IOUtilities.java#L84)):*
> ```Java
>                     return null;
>                 }
>                 InputStream stream = zip.getInputStream(entry);
> ```

Should this be 

    using(var stream = ...)

?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomreader/20)*

**Comment by hatton** [comment]
Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[app/src/main/java/org/sil/bloom/reader/IOUtilities.java, line 84 at r2](https://reviewable.io:443/reviews/bloombooks/bloomreader/20#-Ks9jJF1UtPDYh3_A7fj:-Ks9jgHP_4idRd8frvE0:b-5n9v7p) ([raw file](https://github.com/bloombooks/bloomreader/blob/d35e79051ad8d19cfd8e8e4b7ae9fd7ca4876ba6/app/src/main/java/org/sil/bloom/reader/IOUtilities.java#L84)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

Should this be 

    using(var stream = ...)

?
</blockquote></details>

whoops wrong language!

So instead, should that go in the finally?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomreader/20)*

**Comment by hatton** [comment]
Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[app/src/main/java/org/sil/bloom/reader/IOUtilities.java, line 84 at r2](https://reviewable.io:443/reviews/bloombooks/bloomreader/20#-Ks9jJF1UtPDYh3_A7fj:-Ks9kGNTaPROlg33fCEk:bvx5im) ([raw file](https://github.com/bloombooks/bloomreader/blob/d35e79051ad8d19cfd8e8e4b7ae9fd7ca4876ba6/app/src/main/java/org/sil/bloom/reader/IOUtilities.java#L84)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

whoops wrong language!

So instead, should that go in the finally?
</blockquote></details>

e.g. stream.close();  'fraid I don't know java well enough.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomreader/20)*

**Comment by JohnThomson** [comment]
Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[app/src/main/java/org/sil/bloom/reader/IOUtilities.java, line 84 at r2](https://reviewable.io:443/reviews/bloombooks/bloomreader/20#-Ks9jJF1UtPDYh3_A7fj:-Ks9qWx17MRFXOLIVZQf:b-sk5fb1) ([raw file](https://github.com/bloombooks/bloomreader/blob/d35e79051ad8d19cfd8e8e4b7ae9fd7ca4876ba6/app/src/main/java/org/sil/bloom/reader/IOUtilities.java#L84)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

e.g. stream.close();  'fraid I don't know java well enough.
</blockquote></details>

You're right, of course. Sorry. Fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomreader/20)*

**Pull request merged**